fhir-audit/src/main/java/com/ibm/fhir/audit/logging/mapper/impl/CADFMapper.java

```diff
@@ -0,0 +1,156 @@
+                && logEntry.getContext().getAction() != null
+                && logEntry.getContext().getApiParameters() != null) {
+            // Define resources
+            CadfResource initiator =
+                    new CadfResource.Builder(logEntry.getTenantId() + "@" + logEntry.getComponentId(),
+                            ResourceType.compute_machine)
+                                    .geolocation(
+                                            new CadfGeolocation.Builder(geoCity, geoState, geoCountry, null).build())
+                                    .credential(
+                                            new CadfCredential.Builder("user-" + logEntry.getUserName()).build())
+                                    .host(logEntry.getComponentIp()).build();
+            CadfResource target =
+                    new CadfResource.Builder(
+                            logEntry.getContext().getData() == null || logEntry.getContext().getData().getId() == null
+                                    ? UUID.randomUUID().toString()
+                                    : logEntry.getContext().getData().getId(),
+                            ResourceType.data_database)
+                                    .geolocation(
+                                            new CadfGeolocation.Builder(geoCity, geoState, geoCountry, null).build())
+                                    .address(
+                                            new CadfEndpoint(logEntry.getContext().getApiParameters().getRequest(), "",
+                                                    ""))
+                                    .build();
+
+            FHIRContext fhirContext = new FHIRContext(logEntry.getContext());
+            fhirContext.setClient_cert_cn(logEntry.getClientCertCn());
+            fhirContext.setClient_cert_issuer_ou(logEntry.getClientCertIssuerOu());
+            fhirContext.setEventType(logEntry.getEventType());
+            fhirContext.setLocation(logEntry.getLocation());
+            fhirContext.setDescription(logEntry.getDescription());
+
+            if (logEntry.getContext().getEndTime() == null ||
+                    logEntry.getContext().getStartTime().equalsIgnoreCase(logEntry.getContext().getEndTime())) {
+                cadfEventOutCome = Outcome.pending;
+            } else if (logEntry.getContext().getApiParameters().getStatus() < 400) {
+                cadfEventOutCome = Outcome.success;
+            } else {
+                cadfEventOutCome = Outcome.failure;
+            }
+
+            event = new CadfEvent.Builder(
+                            logEntry.getContext().getRequestUniqueId() == null ? UUID.randomUUID().toString()
+                                    : logEntry.getContext().getRequestUniqueId(),
+                                    EventType.activity, logEntry.getTimestamp(),
+                                    FHIR_TO_CADF.getOrDefault(logEntry.getContext().getAction(), Action.unknown),
```

Comment:
do we ever get an "E" as input to this?  if so, i guess we'll end up setting "Action.unknown" ?

Reply:
`FHIR_TO_CADF.getOrDefault(logEntry.getContext().getAction(), Action.unknown),`
This ends up as an unknown. 

In the  output it looks like this...

Reply:
```
CreateTime:1607706347894
{
    "action": "unknown",
    "eventTime": "2020-12-11 17:05:47.882",
    "eventType": "activity",
    "id": "336794e9-b5f9-4ac5-a00b-72aa7d9c5e0b",
    "outcome": "success",
    "typeURI": "http://schemas.dmtf.org/cloud/audit/1.0/event",
    "tags": [
    ],
    "attachments": [
        {
            "contentType": "application/json",
            "content": "rO0ABXQCFgp7CiAgICAicmVxdWVzdF91bmlxdWVfaWQiOiAiMzM2Nzk0ZTktYjVmOS00YWM1LWEwMGItNzJhYTdkOWM1ZTBiIiwKICAgICJhY3Rpb24iOiAiRSIsCiAgICAic3RhcnRfdGltZSI6ICIyMDIwLTEyLTExIDE3OjA1OjQ3LjUyNCIsCiAgICAiZW5kX3RpbWUiOiAiMjAyMC0xMi0xMSAxNzowNTo0Ny44ODIiLAogICAgImFwaV9wYXJhbWV0ZXJzIjogewogICAgICAgICJyZXF1ZXN0IjogImh0dHBzOi8vbG9jYWxob3N0Ojk0NDMvZmhpci1zZXJ2ZXIvYXBpL3Y0IiwKICAgICAgICAicmVxdWVzdF9zdGF0dXMiOiAyMDAKICAgIH0sCiAgICAiYmF0Y2giOiB7CiAgICAgICAgInJlc291cmNlc19yZWFkIjogMCwKICAgICAgICAicmVzb3VyY2VzX2NyZWF0ZWQiOiAwLAogICAgICAgICJyZXNvdXJjZXNfdXBkYXRlZCI6IDIKICAgIH0sCiAgICAiZXZlbnRfdHlwZSI6ICJmaGlyLWJ1bmRsZSIsCiAgICAiZGVzY3JpcHRpb24iOiAiRkhJUiBCdW5kbGUgcmVxdWVzdCIsCiAgICAibG9jYXRpb24iOiAiMDowOjA6MDowOjA6MDoxL2xvY2FsaG9zdCIKfQ=="
        }
    ],
```